### PR TITLE
WIP: Drop support for Python 2 and yum

### PR DESCRIPTION
**Pull request by ernestask**
One thing I’m not sure about is what to do with the repo table of the database. Add a migration script to replace the type with “dnf”?

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -6,12 +6,11 @@ cat << EOH
 Prepares the source tree for configuration
 
 Usage:
-  autogen.sh [sydeps [--install-yum|--install-dnf]]
+  autogen.sh [sydeps [--install-dnf]]
```
Comment (marusak) [inline autogen.sh]: it would be better to have only --install (as we have in abrt, libreport...)

File: src/bin/faf-migrate-db [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
```
Comment (xsuchy) [inline src/bin/faf-migrate-db]: In Fedora, we prefer /usr/bin/python3 in shebang.

Comment (ernestask) [reply src/bin/faf-migrate-db]: Changed it. Looks like we’ve been sedding these out in the spec file, but I just ripped that out and modified the files directly.

**Comment by xsuchy** [comment]
Lets do the migration in separate PR and merge this one.

**Pull request merged**